packages/rich-text-react-renderer/src/index.tsx

```diff
@@ -0,0 +1,136 @@
+  [BLOCKS.HR]: () => <hr />,
+  [INLINES.ASSET_HYPERLINK]: node => defaultInline(INLINES.ASSET_HYPERLINK, node as Inline),
+  [INLINES.ENTRY_HYPERLINK]: node => defaultInline(INLINES.ENTRY_HYPERLINK, node as Inline),
+  [INLINES.EMBEDDED_ENTRY]: node => defaultInline(INLINES.EMBEDDED_ENTRY, node as Inline),
+  [INLINES.HYPERLINK]: (node, children) => <a href={node.data.uri}>{children}</a>,
+};
+
+const defaultMarkRenderers: RenderMark = {
+  [MARKS.BOLD]: text => <b>{text}</b>,
+  [MARKS.ITALIC]: text => <i>{text}</i>,
+  [MARKS.UNDERLINE]: text => <u>{text}</u>,
+  [MARKS.CODE]: text => <code>{text}</code>,
+};
+
+function defaultInline(type: string, node: Inline) {
+  return (
+    <span key={node.data.target.sys.id}>
+      type: {node.nodeType} id: {node.data.target.sys.id}
+    </span>
+  );
+}
+
+export type CommonNode = Text | Block | Inline;
+
+export interface Next {
+  (nodes: CommonNode[]): ReactNode;
+}
+
+export interface NodeRenderer {
+  (node: Block | Inline, children: Next): ReactNode;
+}
+
+export interface RenderNode {
+  [k: string]: NodeRenderer;
+}
+
+export interface RenderMark {
+  [k: string]: (text: ReactNode) => ReactElement<any>;
+}
+
+export interface Options {
+  /**
+   * Node renderers
+   */
+  renderNode?: RenderNode;
+  /**
+   * Mark renderers
+   */
+  renderMark?: RenderMark;
+}
+
+/**
+ * Serialize a Contentful Rich Text `document` to React tree
+ */
+export function documentToReactTree(
+  richTextDocument: Document,
+  options: Partial<Options> = {},
+): ReactNode {
+  if (!richTextDocument || !richTextDocument.content) {
+    return null;
+  }
+
+  return nodeListToReactTree(richTextDocument.content, {
+    renderNode: {
+      ...defaultNodeRenderers,
+      ...options.renderNode,
+    },
+    renderMark: {
+      ...defaultMarkRenderers,
+      ...options.renderMark,
+    },
+  });
+}
+
+function nodeListToReactTree(nodes: CommonNode[], { renderNode, renderMark }: Options): ReactNode {
+  return nodes.map<ReactNode>((node, index) =>
+    appendKeyToValidElement(nodeToReactTree(node, { renderNode, renderMark }), index),
+  );
+}
+
+function nodeToReactTree(node: CommonNode, { renderNode, renderMark }: Options): ReactNode {
+  if (helpers.isText(node)) {
+    if (node.marks.length > 0) {
+      return node.marks.reduce(
+        (value: ReactNode, mark: Mark) => {
+          if (!renderMark[mark.type]) {
+            return value;
+          }
+          return renderMark[mark.type](value);
+        },
+        node.value as ReactNode,
+      );
+    }
+    return node.value;
+  } else {
+    const children: Next = nodes => nodeListToReactTree(nodes, { renderMark, renderNode });
+    if (!node.nodeType || !renderNode[node.nodeType]) {
+      // TODO: Figure what to return when passed an unrecognized node.
+      return null;
+    }
+    return renderNode[node.nodeType](node, children);
+  }
+}
+
+function appendKeyToValidElement(element: ReactNode, key: string | number): ReactNode {
```

Comment:
```suggestion
function appendKeyToValidElement(element: ReactNode, key: React.Key): ReactNode {
```

Reply:
As @mgmolisani pointed out, this is the internal key that we assign, which is always a number.